src/index.ts

```diff
@@ -53,19 +53,41 @@ export class Status {
         adapter: DbAdapter,
         logger: Logger = CLI_LOGGER
     ): Promise<void> {
-        logger.log('Starting migration\n\n')
-        for (const commit of this.newCommits) {
-            logger.log(`${chalk.yellow.bold(commit.shortSha1)} ${commit.subject}\n`)
-            for (const task of commit.tasks) {
-                logger.log(task.toString() + '...')
-                /* istanbul ignore next */
-                const interval = setInterval(() => logger.log('.'), 100)
-                await task.execute(migrationDir, adapter, this.head, commit)
-                clearInterval(interval)
-                logger.log(' Success\n')
+        while (true) {
+            logger.log(this.toString())
+            const tasks = this.newCommits.reduce<Task[]>((prev, next) => prev.concat(next.tasks), [])
```

Comment:
Why can't this be done anymore in a nested for loop?

Reply:
it is below, but we need all of them before the migrations to create the pending entries

Reply:
wouldn't that be possible with a nested for loop too?